### PR TITLE
Feature/56/choose the time

### PR DESCRIPTION
**Pull request by PaulRomanov**
image

### CODE REVIEW

File: src/app/components/popup-choose-the-time/popup-choose-the-time.component.html [hunk 1]
```diff
@@ -2,10 +2,10 @@
   <div class="wrapper-content">
     <div class="wrapper-data">
       <mat-form-field appearance="fill">
-        <mat-label>Moment.js datepicker</mat-label>
-        <input matInput [matDatepicker]="dp" [formControl]="date">
-        <mat-datepicker-toggle matSuffix [for]="dp"></mat-datepicker-toggle>
-        <mat-datepicker #dp></mat-datepicker>
+        <mat-label>Choose a date</mat-label>
```
Comment (mhasilau) [inline src/app/components/popup-choose-the-time/popup-choose-the-time.component.html]: pls create and use translate service

File: src/app/components/popup-choose-the-time/popup-choose-the-time.component.html [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
+      </mat-form-field> -->
+    </div>
+    <div class="wrapper-time">
+      <app-time *ngFor="let id of times" [id]="id" (delete)="del($event)"></app-time>
     </div>
-    <div class="wrapper-time"></div>
     <div class="wrapper-btn">
       <button class="btn unfilled-btn" type="button">Cancel</button>
-      <button class="btn filled-btn" type="button">OK</button>
+      <button class="btn filled-btn" type="button" (click)="add()">OK</button>
```
Comment (mhasilau) [inline src/app/components/popup-choose-the-time/popup-choose-the-time.component.html]: pls create and use translate service

File: src/app/components/popup-choose-the-time/time/time.component.html [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<label>
+  from
+  <input type="time" />
+  to
+  <input type="time" />
+  <button (click)="removeObject()">del</button>
```
Comment (mhasilau) [inline src/app/components/popup-choose-the-time/time/time.component.html]: pls create and use translate service

File: src/app/components/popup-choose-the-time/popup-choose-the-time.component.html [hunk 4]
```diff
@@ -1,19 +1,26 @@
 <app-modal-window [title]="title">
   <div class="wrapper-content">
     <div class="wrapper-data" [title]="titleData">
+      <span class="title-data">{{ titleData }}</span>
```
Comment (mhasilau) [inline src/app/components/popup-choose-the-time/popup-choose-the-time.component.html:4]: pls create and use translate service, if it doesn't datatime

File: src/app/components/popup-choose-the-time/popup-choose-the-time.component.html [hunk 5]
```diff
@@ -0,0 +1,26 @@
+<app-modal-window [title]="title">
+  <div class="wrapper-content">
+    <div class="wrapper-data" [title]="titleData">
+      <span class="title-data">{{ titleData }}</span>
+      <!-- <mat-form-field appearance="fill">
```
Comment (rastorc3v) [inline src/app/components/popup-choose-the-time/popup-choose-the-time.component.html:5]: remove comments please

File: src/app/components/popup-choose-the-time/popup-choose-the-time.component.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  templateUrl: './popup-choose-the-time.component.html',
+  styleUrls: ['./popup-choose-the-time.component.scss'],
+})
+export class PopupChooseTheTimeComponent {
+  @Input() title: string = 'Choose the time';
+  @Input() titleData: string = '01/11/2021';
+
+  public times: any = [0];
+
+  @ViewChild('container', { read: ViewContainerRef }) container: ViewContainerRef;
```
Comment (rastorc3v) [inline src/app/components/popup-choose-the-time/popup-choose-the-time.component.ts:21]: You don't need in this var anymore. Please clean up

File: src/app/components/popup-choose-the-time/popup-choose-the-time.module.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatSelectModule } from '@angular/material/select';
+import { ModalWindowModule } from '../modal-window/modal-window.module';
+import { MatDatepickerModule } from '@angular/material/datepicker';
+import { MatInputModule } from '@angular/material/input';
+import { OwlDateTimeModule, OwlNativeDateTimeModule } from 'ng-pick-datetime';
+import { BrowserModule } from '@angular/platform-browser';
+import { TimeModule } from './time/time.module';
+
```
Comment (rastorc3v) [inline src/app/components/popup-choose-the-time/popup-choose-the-time.module.ts:13]: do you need in all these modules? verify please

File: src/app/components/popup-choose-the-time/time/time.component.html [hunk 8]
```diff
@@ -0,0 +1,7 @@
+<label>
+  from
+  <input type="time" />
+  to
+  <input type="time" />
+  <button class="btn unfilled-btn btn-minus" (click)="removeObject()">-</button>
+</label>
```
Comment (rastorc3v) [inline src/app/components/popup-choose-the-time/time/time.component.html:7]: please, separate label by two for each input

File: src/app/components/popup-choose-the-time/time/time.component.scss [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+label {
+  margin: 0 25px 0 25px;
+}
+
+.wrapper-time {
+  margin: 15px 0 0 0;
+}
+
+input[type='time' i] {
```
Comment (rastorc3v) [inline src/app/components/popup-choose-the-time/time/time.component.scss:16]: why do you add 'i' here?

**Review by rastorc3v — CHANGES_REQUESTED** [review]

File: src/app/components/popup-choose-the-time/time/time.component.scss (hunk 9 above)
Comment (PaulRomanov) [reply src/app/components/popup-choose-the-time/time/time.component.scss:16]: fix

File: src/app/components/popup-choose-the-time/popup-choose-the-time.module.ts (hunk 7 above)
Comment (PaulRomanov) [reply src/app/components/popup-choose-the-time/popup-choose-the-time.module.ts:13]: removed unnecessary

File: src/app/components/popup-choose-the-time/popup-choose-the-time.component.html (hunk 5 above)
Comment (PaulRomanov) [reply src/app/components/popup-choose-the-time/popup-choose-the-time.component.html:5]: removed

File: src/app/components/popup-choose-the-time/time/time.component.html (hunk 8 above)
Comment (PaulRomanov) [reply src/app/components/popup-choose-the-time/time/time.component.html:7]: fix

File: src/app/components/popup-choose-the-time/popup-choose-the-time.component.html (hunk 1 above)
Comment (PaulRomanov) [reply src/app/components/popup-choose-the-time/popup-choose-the-time.component.html]: add

File: src/app/components/popup-choose-the-time/time/time.component.html (hunk 3 above)
Comment (PaulRomanov) [reply src/app/components/popup-choose-the-time/time/time.component.html]: add

File: src/app/components/popup-choose-the-time/popup-choose-the-time.component.html (hunk 4 above)
Comment (PaulRomanov) [reply src/app/components/popup-choose-the-time/popup-choose-the-time.component.html:4]: add

File: src/app/components/popup-choose-the-time/popup-choose-the-time.component.html (hunk 2 above)
Comment (PaulRomanov) [reply src/app/components/popup-choose-the-time/popup-choose-the-time.component.html]: add

**Review by mhasilau — APPROVED** [review]

File: src/app/components/popup-choose-the-time/popup-choose-the-time.component.ts [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    // eslint-disable-next-line
+    this.times.push(lastId + 1);
+  }
+  // eslint-disable-next-line
+  del(timeId: number) {
+    this.times = this.times.filter((id: number) => {
+      return id !== timeId;
+    });
+  }
+}
```
Comment (Nicolai-boiko) [inline src/app/components/popup-choose-the-time/popup-choose-the-time.component.ts:34]: Why all functions with disabled eslint?

File: src/app/components/popup-choose-the-time/popup-choose-the-time.component.ts [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+  templateUrl: './popup-choose-the-time.component.html',
+  styleUrls: ['./popup-choose-the-time.component.scss'],
+})
+export class PopupChooseTheTimeComponent {
+  @Input() title: string = 'Choose the time';
+  @Input() titleData: string = '01/11/2021';
+
+  public times: any = [0];
+
+  // eslint-disable-next-line
```
Comment (vstrabko) [inline src/app/components/popup-choose-the-time/popup-choose-the-time.component.ts:17]: Eslint should be enabled and all types returned by functions should be in place.
This one should return `void`.

**Review by vstrabko — CHANGES_REQUESTED** [review]

File: src/app/components/popup-choose-the-time/popup-choose-the-time.component.ts (hunk 11 above)
Comment (PaulRomanov) [reply src/app/components/popup-choose-the-time/popup-choose-the-time.component.ts]: I fixed it

File: src/app/components/popup-choose-the-time/popup-choose-the-time.component.ts (hunk 10 above)
Comment (PaulRomanov) [reply src/app/components/popup-choose-the-time/popup-choose-the-time.component.ts:31]: I fixed it

File: src/app/components/popup-choose-the-time/time/time.component.ts [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+@Component({
+  selector: 'app-time',
+  templateUrl: './time.component.html',
+  styleUrls: ['./time.component.scss'],
+})
+export class TimeComponent {
+  @Input() id: number;
+  @Output() delete = new EventEmitter<number>();
+  // eslint-disable-next-line
+  removeObject() {
```
Comment (vstrabko) [inline src/app/components/popup-choose-the-time/time/time.component.ts:13]: same thing about types here.

**Review by vstrabko — CHANGES_REQUESTED** [review]

File: src/app/components/popup-choose-the-time/time/time.component.ts (hunk 12 above)
Comment (PaulRomanov) [reply src/app/components/popup-choose-the-time/time/time.component.ts]: fix

**Review by vstrabko — APPROVED** [review]

**Review by rastorc3v — APPROVED** [review]

**Pull request merged**